pco/src/data_types/floats.rs

```diff
@@ -147,7 +150,121 @@ macro_rules! impl_float_number {
         l as Self
       }
     }
+  };
+}
+
+impl FloatLike for f16 {
+  const BITS: Bitlen = 16;
+  const PRECISION_BITS: Bitlen = Self::MANTISSA_DIGITS as Bitlen - 1;
+  const ZERO: Self = f16::ZERO;
+  const MAX_FOR_SAMPLING: Self = f16::from_bits(30719); // Half of MAX size.
+
+  #[inline]
+  fn abs(self) -> Self {
+    Self::from_f32(self.to_f32().abs())
+  }
+
+  fn inv(self) -> Self {
+    Self::from_f32(1.0 / self.to_f32())
```

Comment:
I think these `from_f32(... self.to_f32())` ones can be rewritten to use native f16 ops when available. Like I think this one could be `Self::ONE / self`.

Reply:
Probably, though I'm not sure what exactly will happen at the lower level since I'm not sure the CPU has a native `f16` divide op. I just copied the [implementation from `half`](https://docs.rs/half/latest/src/half/num_traits.rs.html#269), but I'll change this to `Self::ONE / self`.

Got any suggestions from the others? Again I just copied from `half`:
https://docs.rs/half/latest/src/half/num_traits.rs.html#206
https://docs.rs/half/latest/src/half/num_traits.rs.html#436